### ISSUE TITLE
Grade item

### ISSUE THREAD

**Issue by xow** [xow/moodle-mod_quizgame#12]
It's unlikely that a quizventure game will contribute to a student's final score, but it's likely that a teacher may want to add it (with 0 weight) to the user's course report. So a grade item could be useful. Will need to work out some kind of maximum grade.

**Comment by xow** [comment]
Possible solution: limit the max grade to the maximum possible score if you get all the questions right. Instead of having the speed go up once you go through all the questions, have it end. Then make your grade out of that maximum score.

**Comment by xow** [comment]
Added a grade field to the DB in 2e6d5f9. This fixes a small bug when renaming. Still needs a formula to convert score to grade, and implementation

**Comment by xow** [comment]
Idea 2: Score = how many questions you got correct divided by how many you faced or total different questions whichever is bigger

score = correct/MAX(faced, total questions)

Problem: if you get every question right you don't get any benefit from keeping on playing after the first round. Your best bet is to die before round 2

**Comment by rschrenk** [comment]
Hi,

Quizventure is a great plugin but the lack of grading is a pitty! It would be great if you find a solution to this. As always with moodle - there will not be the "one" solution, but you will need some choices. What about:

1. Score Paced: Teacher sets a target score and it has to be reached somehow
2. Time Paced: Teacher sets a time limit and student can try to reach as much score as possible. The grading is a percentage of the score compared to the max-score (maximum 100%)
3. Average: Student just gets his average score
4. Average + Minimum-Questions: Student has to answer at least x Questions. After that point he can quit at any time getting his average score. That way the teacher requires the student to answer at least a minimum amount of questions and then the student can improve his grades.

Kind regards

**Comment by xow** [comment]
These are great ideas, thanks

Number 3 (the average) could be tricky if they get a couple of bad runs they need to offset it with lots of really good ones

**Comment by rschrenk** [comment]
Thats true, but the same as with the standard quiz module. But turns out not to be a problem if you can trigger "best result", "average result" and so on for a series of attempts. Thats how it is done in the quiz module.

**Comment by xow** [comment]
Excellent poin @rschrenk
